### ISSUE TITLE
[oracle] 'Found unparsable section' error with (+) and LT02 violation

### ISSUE THREAD

**Issue by osarier** [sqlfluff/sqlfluff#7940]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I wanted to run the linter on a file with the following example. This leads to the described error. 

### Expected Behaviour

I would expect the example to be correctly parsed and no LT02 violation. 

### Observed Behaviour

I observed the following output:
>L:   4 | P:  15 | LT02 | Expected line break and no indent before '('.
                       | [layout.indent]
L:   4 | P:  16 |  PRS | Line 4, Position 16: Found unparsable section: "(+) =
                       | 'smth'"

### How to reproduce

```sql
select e1.abc from exp1 e1, exp2 e2
where
    e1.key (+) = e2.key
    and e1.str (+) = 'smth';
``` 

### Dialect

oracle

### Version

4.2.1
python 3.11.3

### Configuration

[sqlfluff]
dialect = oracle

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
Hi @osarier  - thanks for opening this PR.

I have looked at the official [Oracle JOIN definition](https://docs.oracle.com/en/database/oracle/oracle-database/26/sqlrf/Joins.html), and they are explicitely saying that 
> Oracle recommends that you use the FROM clause OUTER JOIN syntax rather than the Oracle join operator.

In that sense, while Oracle might still supports the (+) operator for backward compatibility, Oracle itself has recommended using the standard LEFT/RIGHT JOIN syntax as far as I know for over two decades, so I'm hesitant to add features in `sqlfluff` to enable this apparently discouraged practice.

@saulotoledo - what is your POV?

**Comment by osarier** [comment]
Hi @peterbud 
thanks for looking into this Issue!

I understand where you are coming from.

The thing is, this feature is already in there. The following example works just fine:
```sql
select e1.abc from exp1 e1, exp2 e2
where
   e1.key (+) = e2.key
   and e1.str (+) = e2.str;
``` 

So in my eyes there is just a broken aspect of the feature which needs to be fixed.

**Comment by saulotoledo** [comment]
Hi @peterbud. You make a good point, and it got me thinking. On the other hand, SQLFluff already has some existing support for `(+)` (e.g., [here](https://github.com/sqlfluff/sqlfluff/blob/5f6869f4047ae6816e9dd338b1fac458eba45bca/test/fixtures/dialects/oracle/non_ansi_joins.sql) and as mentioned by @osarier), so this currently feels like broken functionality rather than an unsupported feature.

Since `(+)` is an Oracle-proprietary syntax, it might not be removed anytime soon, even if it is discouraged. Because of that, I still think it's worth adding the fix.

**Comment by peterbud** [comment]
OK, I see - I agree, let's try to fix it.
